src/psyclone/tests/psyGen_test.py

```diff
@@ -1392,21 +1393,76 @@ def test_argument_backward_dependence(monkeypatch, annexed):
 
 
 def test_node_depth():
-    '''Test that the Node class depth method returns the correct value
-    for a Node in a tree '''
+    '''
+    Test that the Node class depth method returns the correct value for a
+    Node in a tree. The start depth to determine a Node's depth is set to
+    0. Depth of a Schedule is 1 and increases for its descendants.
+    '''
     _, invoke_info = parse(
         os.path.join(BASE_PATH, "1_single_invoke.f90"),
         distributed_memory=True, api="dynamo0.3")
     psy = PSyFactory("dynamo0.3", distributed_memory=True).create(invoke_info)
     invoke = psy.invokes.invoke_list[0]
     schedule = invoke.schedule
+    # Assert that start_depth of any Node (including Schedule) is 0
+    assert schedule.START_DEPTH == 0
+    # Assert that Schedule depth is 1
     assert schedule.depth == 1
+    # Depth increases by 1 for descendants at each level
     for child in schedule.children:
         assert child.depth == 2
     for child in schedule.children[3].children:
         assert child.depth == 3
 
 
+def test_node_position(tmpdir, f90, f90flags):
+    '''
+    Test that the Node class position and abs_position methods return
+    the correct value for a Node in a tree. The start position is
+    set to 0. Relative position starts from 0 and absolute from 1.
+    '''
+    _, invoke_info = parse(
+        os.path.join(BASE_PATH, "4.7_multikernel_invokes.f90"),
+        distributed_memory=True, api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=True).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+    child = schedule.children[6]
+    # Assert that start_position of any Node is 0
+    assert child.START_POSITION == 0
+    # Assert that relative and absolute positions return correct values
+    assert child.position == 6
+    assert child.abs_position == 7
+    # Test InternalError for _find_position
+    with pytest.raises(InternalError) as excinfo:
+        _, _ = child._find_position(child.root.children, -2)
+    assert "started from -2 instead of 0" in str(excinfo.value)
+
+    if TEST_COMPILE:
```

Comment:
One of the development principles we have is to ultimately test all F90 examples for compilation and this is currently not done for "4.7_multikernel_invokes.f90", so I tested it here since I modified the test. I moved the compilation test to one of two tests in `dynamo0.3_transformations.test` that use this example.

Reply:
That's above and beyond - thanks!